packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -398,6 +442,14 @@ class BouncingScrollPhysics extends ScrollPhysics {
   @override
   double applyBoundaryConditions(ScrollMetrics position, double value) => 0.0;
 
+  @override
+  double adjustPositionForNewDimensions(ScrollMetrics position, {
+    @required bool isScrolling,
+    @required double velocity,
+  }) {
+    return position.pixels.clamp(position.minScrollExtent, position.maxScrollExtent) as double;
```

Comment:
this is the main thing from my patch that is going to need more work -- specifically, we need to handle the case of being overscrolled in a more graceful manner.

Ideally what I'd love for us to do is if we are under/over scrolled, we maintain the delta from the min/max extent to the position before and after the correction. The difficulty is that here we're not passing in the old min/max extent. Maybe we could just pass them in? I guess we have them in applyContentDimensions which is what is calling this...

cc @goderbauer

Reply:
maybe we pass in two ScrollMetrics, one from before the update and one from after?
If we do this then we should probably avoid calling this method in the case where the old metrics were null...

Reply:
Check out my widget test for resizing while dragging.  I think that captures what we were saying yesterday about keeping a valid drag going.